pytorch_lightning/overrides/torch_distributed.py

```diff
@@ -15,7 +14,10 @@
 logger = logging.getLogger(__name__)
 
 if torch.distributed.is_available():
+    from torch._C._distributed_c10d import ProcessGroup
     from torch.distributed import Backend, broadcast, get_backend, get_rank, GroupMember
+else:
+    ProcessGroup = None
```

Comment:
ProcessGroup only gets imported for typing. Wouldn't the type check fail if the else branch is considered, as None is not a valid type annotation.

Reply:
I think `None` is a valid type annotation. Also, it doesn't error out on my system when `torch.distrubted.is_available()` is `False`. (source: https://stackoverflow.com/a/38516083)

```python
>>> def nothing(nun: None) -> None:
...     return nun
... 
>>> nothing(None)
>>> 
```

Reply:
I think `None` may be too strict in this case. Should it be `object` instead? @awaelchli @krshrimali 
```suggestion
else:
    ProcessGroup = object
```

Reply:
Thanks, @akihironitta for taking another look. I was just reading about it on the internet and found this interesting answer: https://stackoverflow.com/a/36286947. Apparently, this fixes for my environment (which doesn't have `distributed` module in PyTorch). What do you think? (just pushed a commit with my proposed solution)